core/app/models/spree/shipment.rb

```diff
@@ -287,13 +269,6 @@ def update_attributes_and_order(params = {})
           order.updater.update_shipment_total
           order.updater.update_payment_state
 
-          # Update shipment state only after order total is updated because it
-          # (via Order#paid?) affects the shipment state (YAY)
-          self.update_columns(
-            state: determine_state(order),
```

Comment:
here is the other place that `determine_state` is being called which is triggered when `Shipments#update` is hit. However I believe that a shipment's state should only be changed due to signifyd or quiet, not by the shipment itself being updated or by us outside of the state machine.